### ISSUE TITLE
Delete via reference currently requires download first

### ISSUE THREAD

**Issue by BSVogler** [beda-software/fhir-py#103]
given a Patient id I want to delete it from the server. This code is required to make it work.

```python
client.reference('Patient', str(userid)).to_resource().delete()
```

The code above downloads the resource first, although it is not necessary. It would be good to have 
```python
client.reference('Patient', str(userid)).delete()
```

instead.

**Comment by BSVogler** [comment]
Might be related with this https://github.com/beda-software/fhir-py/issues/101

**Comment by ruscoder** [comment]
@pavlushkin please take a look, it's related to refactoring you're making for conditional operations

**Comment by ruscoder** [comment]
Will be fixed in 2.0.0 using `client.delete('Patient', 'id')` or `client.delete('Patient/id')`

**Comment by ruscoder** [comment]
Also added helpers for BaseReference to use `.delete()` and `.patch(**kwargs)` as it was requested in the issue
